types/types.go

```diff
@@ -207,6 +211,10 @@ type UnparsedImage interface {
 	Manifest() ([]byte, string, error)
 	// Signatures is like ImageSource.GetSignatures, but the result is cached; it is OK to call this however often you need.
 	Signatures(ctx context.Context) ([][]byte, error)
+	// UpdatedLayerInfos returns either nil (meaning there are no updates), or updated values for the layer blobsums that are listed in the image's manifest.
+	// The Digest field is guaranteed to be provided, Size may be -1 and MediaType may be optionally provided.
+	// WARNING: The list may contain duplicates, and they are semantically relevant.
+	UpdatedLayerInfos() []BlobInfo
```

Comment:
FIXME

At a first glance this seems risky; it allows `Manifest()` to be validated against `Signatures()`, but the user will be encouraged to ignore the validated data and use `UpdatedLayerInfos()` instead. (The `!ic.canModifyManifest` check in `copy.Image` does prevent this from happening currently, but that’s rather non-obvious.)

#288 instead introduces `ImageSource.GetOriginalManifest` etc., forcing the caller to make a choice in advance between getting the original manifest (which may be unusable for reading data) and getting the current one (which may break signatures).  I’m not at all sure that is better, though.

Reply:
Note to self: If the same per-transport policy is used for copying out of the storage, and for run decisions within, what does that mean for the original vs. modified manifest?

Reply:
I think we'd need some way to affect which variation `copy.Image()` asked for, but otherwise I think that would work.

Reply:
> #288 instead introduces ImageSource.GetOriginalManifest etc., forcing the caller to make a choice in advance between getting the original manifest (which may be unusable for reading data) and getting the current one (which may break signatures). I’m not at all sure that is better, though.

@mtrmac is it possible to extract `GetOriginalManifest` to another new PR with just that so we can merge it and reuse it here for @nalind ? (or carry the patch here directly)